src/test/java/com/faforever/client/io/DownloadServiceTest.java

```diff
@@ -0,0 +1,167 @@
+package com.faforever.client.io;
+
+import com.faforever.client.preferences.PreferencesService;
+import com.faforever.client.test.ServiceTest;
+import com.faforever.commons.io.ByteCountListener;
+import javafx.collections.FXCollections;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class DownloadServiceTest extends ServiceTest {
+
+  @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+  private PreferencesService preferencesService;
```

Comment:
There is no need to use a deep stub here. We have a preferencesBuilder for testing so we can pass complete data